next/components/signup/Auth/index.js

```diff
@@ -33,35 +35,34 @@ const Auth = () => {
     // 랜딩에서 이메일 입력 후 버튼을 눌렀다면, 해당 이메일을 자동 입력
     if (landingEmail === null) return;
 
-    emailEl.current.value = landingEmail;
+    setEmail(landingEmail);
     dispatch(saveLandingEmail(null));
-  }, [landingEmail, dispatch]);
+  }, [landingEmail, setEmail, dispatch]);
 
   useEffect(() => {
-    const { success, email } = router.query;
+    const { success, email: verifiedEmail } = router.query;
 
     const isQuery = Boolean(success);
     if (!isQuery) {
       return;
     }
 
     if (success === 'true') {
-      dispatch(verifyEmailRequest(email));
+      dispatch(verifyEmailRequest(verifiedEmail));
```

Comment:
```suggestion
      dispatch(verifyEmailRequest(verifiedEmail));
```
![image](https://user-images.githubusercontent.com/50310464/120935562-5a311d00-c73e-11eb-960f-52146673ef8b.png)

Reply:
``` js
const [value, setValue] = useState('');
if (...) {
  alert(value)
}
```
위 코드가 가능한 것과 같은 맥락인 듯 하네요. useDispatch를 if문 안에서 사용했다면 오류였고 eslint부터가 이미 잡아냈을 테지만, 전 if문 안에서 훅스를 쓴게 아니라 훅스의 리턴값을 쓴겁디당

Reply:
확인!

Reply:
님 잠결에 리뷰했져?

Reply:
근데 쿼리로 온 이메일 재검증이니 verifiedEmail 보단 queryEmail이 맞겠네요 수정해야겠다 감사합니당

Reply:
dispatch 는
` const dispatch = useDispatch();`
이렇게 끌어오는것 아닌가요?

Reply:
맞져 그것또한 훅스가 아닌 훅스의 리턴값

Reply:
https://stackoverflow.com/questions/58543102/using-reduxs-usedispatch-in-a-custom-hook-yields-an-error

Reply:
훅스의 리턴값이 훅스일수 있어서요

Reply:
dispatch는 if 문 내부에서 사용해도 되는 것 같습니다 감사합니다 ㅎ

Reply:
https://stackoverflow.com/questions/46810526/react-redux-dispatch-2-actions-based-on-condition

Reply:
https://www.digitalocean.com/community/tutorials/redux-redux-thunk

Reply:
훅스의 리턴이 훅스일수도 있군요 덕분에 알아갑니다